src/Synth/ADnote.cpp

```diff
@@ -908,14 +1072,27 @@ void ADnote::initparameters(WatchManager *wm, const char *prefix)
                || (vce.FMEnabled == FMTYPE::RING_MOD))
                 tmp = getFMvoicebasefreq(nvoice);
 
-            if(!pars.GlobalPar.Hrandgrouping)
-                pars.VoicePar[vc].FmGn->newrandseed(prng());
+            if(!waveTables) // debug code, testing only
+                if(!pars.GlobalPar.Hrandgrouping)
+                    pars.VoicePar[vc].FmGn->newrandseed(prng());
 
             for(int k = 0; k < vce.unison_size; ++k)
-                vce.oscposhiFM[k] = (vce.oscposhi[k]
-                                         + pars.VoicePar[vc].FmGn->get(
-                                             vce.FMSmp, tmp))
-                                        % synth.oscilsize;
+            {
+                if(waveTables)
+                {
+                    assert(pars.VoicePar[nvoice].tableMod); // did you allocate ADnoteParameters and not assign its "table" member?
```

Comment:
Same thing as elsewhere: should there be multiple copy operations when unison_size != 1?